skyportal/handlers/api/reminder.py

```diff
@@ -0,0 +1,900 @@
+                    .all()
+                )
+            elif associated_resource_type.lower() == "shift":
+                reminders = (
+                    ReminderOnShift.query_records_accessible_by(self.current_user)
+                    .filter(ReminderOnShift.shift_id == resource_id)
+                    .all()
+                )
+            else:
+                return self.error(
+                    f'Unsupported associated resource type "{associated_resource_type}".'
+                )
+            self.verify_and_commit()
+            return self.success(data=reminders)
+
+        try:
+            reminder_id = int(reminder_id)
+        except (TypeError, ValueError):
+            return self.error("Must provide a valid (scalar integer) reminder ID. ")
+
+        # the default is to reminder on an object
+        if associated_resource_type.lower() == "sources":
+            try:
+                reminder = Reminder.get_if_accessible_by(
+                    reminder_id, self.current_user, raise_if_none=True
+                )
+            except AccessError:
+                return self.error(
+                    'Could not find any accessible reminders.', status=403
+                )
+            reminder_resource_id_str = str(reminder.obj_id)
+
+        elif associated_resource_type.lower() == "spectra":
+            try:
+                reminder = ReminderOnSpectrum.get_if_accessible_by(
+                    reminder_id, self.current_user, raise_if_none=True
+                )
+            except AccessError:
+                return self.error(
+                    'Could not find any accessible reminders.', status=403
+                )
+            reminder_resource_id_str = str(reminder.spectrum_id)
+        elif associated_resource_type.lower() == "gcn_event":
+            try:
+                reminder = ReminderOnGCN.get_if_accessible_by(
+                    reminder_id, self.current_user, raise_if_none=True
+                )
+            except AccessError:
+                return self.error(
+                    'Could not find any accessible reminders.', status=403
+                )
+            reminder_resource_id_str = str(reminder.gcn_id)
+        elif associated_resource_type.lower() == "shift":
+            try:
+                reminder = ReminderOnShift.get_if_accessible_by(
+                    reminder_id, self.current_user, raise_if_none=True
+                )
+            except AccessError:
+                return self.error(
+                    'Could not find any accessible reminders.', status=403
+                )
+            reminder_resource_id_str = str(reminder.shift_id)
+        # add more options using elif
+        else:
+            return self.error(
+                f'Unsupported associated_resource_type "{associated_resource_type}".'
+            )
+
+        if reminder_resource_id_str != resource_id:
+            return self.error(
```

Comment:
Can't you do that at the same time as you query the reminders instead? I'm not sure that this should be an error. Doesn't it just mean that there is no reminder for the resource ?

Reply:
This is a holdover from the comments stuff.